kv/dist_sender_server_test.go

```diff
@@ -98,23 +98,39 @@ func setupMultipleRanges(t *testing.T) (*server.TestServer, *client.DB) {
 	return s, db
 }
 
-// TestMultiRangeScan verifies operation of a scan across ranges.
-func TestMultiRangeScan(t *testing.T) {
+// TestMultiRangeScan verifies that Scan, DeleteRange and ResolveIntentRange
+// work across ranges.
+func TestMultiRangeScanDeleteResolve(t *testing.T) {
 	defer leaktest.AfterTest(t)
-	s, db := setupMultipleRanges(t)
+	s, db := setupMultipleRanges(t, "b")
 	defer s.Stop()
 
-	// Write keys "a" and "b".
-	for _, key := range []string{"a", "b"} {
+	// Write keys before, at, and after the split key.
+	for _, key := range []string{"a", "b", "c"} {
 		if err := db.Put(key, "value"); err != nil {
 			t.Fatal(err)
 		}
 	}
-
-	if rows, err := db.Scan("a", "c", 0); err != nil {
-		t.Fatalf("unexpected error on scan: %s", err)
-	} else if l := len(rows); l != 2 {
-		t.Errorf("expected 2 rows; got %d", l)
+	// Scan to retrieve the keys just written.
+	if rows, err := db.Scan("a", "q", 0); err != nil {
+		t.Fatalf("unexpected error on Scan: %s", err)
+	} else if l := len(rows); l != 3 {
+		t.Errorf("expected 3 rows; got %d", l)
+	}
+	// Delete the keys within a transaction. Implicitly, the intents are
+	// resolved via ResolveIntentRange upon completion.
+	if err := db.Txn(func(txn *client.Txn) error {
```

Comment:
That would do the same because it forces the call to be wrapped in a transaction and re-executed since it spans ranges, but I prefer having the transaction explicit here because I care about intents being created. Plus the wrapping already happens in the above `Scan`, so this adds variety.